scripts/calculateBundleSize.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 - present Instructure, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+const esbuild = require('esbuild')
+const path = require('path')
+const { promisify } = require('util')
+const { exec } = require('child_process')
+const fsp = require('fs/promises')
+const execAsync = promisify(exec)
+
+const esBuild = async (entryPoint, sha) => {
+  const outFile = `./.tmp/${sha}_size_bundle.js`
+  const result = await esbuild.build({
+    entryPoints: [entryPoint],
+    bundle: true,
+    outfile: outFile,
+    metafile: true,
+    external: ['fs', 'path', 'os', 'module'],
+    treeShaking: true
+  })
+
+  const analysis = await esbuild.analyzeMetafile(result.metafile)
+
+  return { bundle: result, outFile, analysis }
+}
+const gzip = async (fileName) => {
+  const file = path.resolve(fileName)
+  await execAsync(`gzip -k ${file}`)
+
```

Comment:
Oh, this looks nasty! (Didn't even know GitHub can do this...)

Reply:
Well, when we run a github action essentially we get a fully fletched linux at our hands, so it is not GitHub that does this rather the underlying linux vm.
And it might look like a security issue (which it is) but this kind of thing is in the same as the other [(ReDos)](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS#) security vulnerabilities. Basically if an attacker somehow gets access to our source code then, he/she may be able to shut down our CI vm's.